prover/protocol/compiler/lookup/compiler.go

```diff
@@ -108,6 +114,10 @@ func CompileLogDerivative(comp *wizard.CompiledIOP) {
 		va.Name = zC.Name
 	}
 
+	if comp.HasSeed {
```

Comment:
I don't understand what comp.HasSeed has to do with how the lookup compiler works. The only important thing is how the coin are sampled but then this logic should be implemented in the `coin` package.